src/iceberg/snapshot.cc

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "iceberg/snapshot.h"
+
+#include <format>
+
+#include "iceberg/util/formatter.h"
+
+namespace iceberg {
+
+std::optional<std::string> Snapshot::operation() const {
+  auto it = summary.find(std::string(SnapshotSummaryFields::kOperation));
```

Comment:
(I thought modern C++ allowed you to index with string_view directly?)

If we're going to be wrapping with `std::string` all the time maybe let's just make them actual `static std::string`?

Reply:
What about `inline static const std::string kFoo = "bar";` to avoid defining these in the .cc file?

Reply:
good point, will change.